src/cpp/include/openvino/genai/text_streamer.hpp

```diff
@@ -13,29 +13,35 @@ using CallbackTypeVariant = std::variant<bool, StreamingStatus>;
 
 /**
  * @brief TextStreamer is used to decode tokens into text and call a user-defined callback function.
- * 
+ *
  * @param tokenizer Tokenizer object to decode tokens into text.
- * @param callback User-defined callback function to process the decoded text, callback should return 
+ * @param callback User-defined callback function to process the decoded text, callback should return
  * either boolean flag or StreamingStatus.
  */
-class OPENVINO_GENAI_EXPORTS TextStreamer: public StreamerBase {
-    StreamingStatus set_streaming_status(CallbackTypeVariant callback_status);
-
-    std::function<CallbackTypeVariant(std::string)> m_subword_callback = [](std::string words)->bool { return false; };
-    StreamingStatus run_callback_if_needed(const std::string& text);
-
+class OPENVINO_GENAI_EXPORTS TextStreamer : public StreamerBase {
 public:
     StreamingStatus write(int64_t token) override;
+    StreamingStatus write(const std::vector<int64_t>& tokens) override;
```

Comment:
FYI @mzegla
as you requested such method

Reply:
Yes, that should remove the need for our WA for streaming the prompt back. Thanks.